### PR TITLE
82 auto update

### PR DESCRIPTION
**Pull request by stefanrinderle**
Auto reload every 5 minutes implemented

* New SQ service for component info
* New auto reload service to set the current analysisDate
* New app reaction to react on changes of the analysis date -> reload data

### CODE REVIEW

File: softvis3d-frontend/src/services/sonarqube/SonarQubeComponentInfoService.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    constructor(projectKey: string, appStatusStore: AppStatusStore, baseUrl?: string) {
+        super(baseUrl);
+
+        this.projectKey = projectKey;
+        this.appStatusStore = appStatusStore;
+    }
+
+    public loadComponentInfo(): Promise<SonarQubeComponentInfo> {
+        return new Promise<SonarQubeComponentInfo>((resolve, reject) => {
+            const params = {component: this.projectKey};
```
Comment (Ungolianth) [inline softvis3d-frontend/src/services/sonarqube/SonarQubeComponentInfoService.ts]: > ## GET api/components/show
> [...] The 'id' or 'key' must be provided.

This should be `{key: this.projectKey}`. At least with SonarQube Version 6.3.1.

File: softvis3d-frontend/src/services/AutoReloadService.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        this.appStatusStore = appStatusStore;
+        this.componentInfoService = componentInfoService;
+    }
+
+    public startAutoReload(): void {
+        if (this.timer) {
+            window.clearInterval(this.timer);
+        }
+
+        this.timer = window.setInterval(this.loadComponentInfo, AutoReloadService.RELOAD_INTERVAL_MS);
```
Comment (Ungolianth) [inline softvis3d-frontend/src/services/AutoReloadService.ts]: When testing the feature I had problems with the unbound "this", at execution time.

Either of those would take care of the problem:
```js
window.setInterval(() => this.loadComponentInfo(), AutoReloadService.RELOAD_INTERVAL_MS)
window.setInterval(this.loadComponentInfo.bind(this), AutoReloadService.RELOAD_INTERVAL_MS)
```

**Comment by Ungolianth** [comment]
Found another issue: The sidebar breaks after a reload. (I believe it may still be bound to the old canvas)?
To recreate:
 * change reload time to 10 seconds
 * load city
 * click on element in canvas (eg. a street)
 * wait for reload
 * select an element in the sidebar
==> Nothing happens

After clicking on the canvas the sidebar works again.

File: softvis3d-frontend/src/services/AutoReloadService.ts (hunk 2 above)
Comment (stefanrinderle) [reply softvis3d-frontend/src/services/AutoReloadService.ts]: ups. thanks. is fixed now.

File: softvis3d-frontend/src/services/sonarqube/SonarQubeComponentInfoService.ts (hunk 1 above)
Comment (stefanrinderle) [reply softvis3d-frontend/src/services/sonarqube/SonarQubeComponentInfoService.ts]: This one is funny. They changed the request parameter of the api/components/show webservice with SQ 6.4. Description in 

6.3.1: "The 'id' or 'key' must be provided."
6.4: "The 'componentId' or 'component' parameter must be provided."

I took the time to check with which revision / issue this change happened and it was commited with this revision: 
https://github.com/SonarSource/sonarqube/commit/a914082d1e284d97f6d1d632ab04e8ba4eff30f7

Unfortunately, there is no more SONAR-8774 as it says in the commit message:
https://jira.sonarsource.com/browse/SONAR-8774

And i could not find any mentioning of that in the release noted or change logs. Sometimes its really funny how they just change stuff.

To make it simple, i changed the implementation that it does not break 6.3.1. As the parameter analysisDate is also only available from 6.4, the auto reload feature is now disabled in 6.3.1.

**Comment by stefanrinderle** [comment]
Fixed all open issues from the review.

**Comment by stefanrinderle** [comment]
No response within 14 days. Will be merged.

**Pull request merged**